Pidaq/Lib/database/database.py

```diff
@@ -0,0 +1,194 @@
+            formatted_measurements = format_data(measurements)
+            data = self.client.query(
+                f'SELECT {formatted_tags} FROM {formatted_measurements}',
+                database=self.current_database
+            )
+        else:
+            return -1 # Must specify at least a tag or a measurement
+        try:
+            return data.raw['series']
+        except KeyError:
+            return -1
+
+    def create_retention_policy(self, name: str, duration: str, replication: str) -> None:
+        """Creates a retention policy for the current database"""
+
+        self.client.create_retention_policy(
+            name,
+            duration,
+            replication,
+            database=self.current_database,
+            default=True
+        )
+
+    def export_to_csv(self, test_name: str, tags=None, measurements=None, csv_path=None) -> None:
+        """Exports measurements to a csv file
+
+        Args:
+            test_name (str): Name of test, becomes name of csv file
+            tags (list(str)): Tags and fields to be included in csv
+            measurements (list(str)): Measurements to be included in csv
+            csv_path(str): File path of where the csv will be written
+        """
+
+        file_name = f'{test_name}.csv'
+        data = self.read_data(tags=tags, measurements=measurements)
+        if csv_path is None:
+            csv_path = os.getcwd()
+        with open(f'{csv_path}\\{file_name}', 'w', newline='\n') as csvfile:
+            writer = csv.writer(csvfile, delimiter=',')
+            writer.writerow(data[0]['columns'])
+            for measurement in data:
+                for row in measurement['values']:
+                    writer.writerow(row)
+
+
+def format_data(data: list) -> str:
+    """Formats data to be able to utilize in the Influx query
+
+    Args:
+        data(list(str)): List of data to be formatted
+
+    Returns formatted_data(str): comma separated data in format
+                                 '"data1","data2","data3"...'
+    """
+
+    list_data = []
+    for point in data:
+        list_data.append(f'"{point}"')
+    formatted_data = ','.join(list_data)
+    return formatted_data
+
+if __name__ == "__main__":
+    # Testing class functionality
+
+    DATABASE = Influx('example')
+
+    # Print data from pressure measurement in formatted json
+    print(DATABASE.read_data(tags=['millibar', 'Pascal'], measurements=['pressure']))
+
+    # Export example data to a csv
+    CWD = os.getcwd()
+    DATABASE.export_to_csv('test', measurements=['pressure', 'temperature'])
+
+    # Log new data to the database, and print to console
+    FIELDS = {
+        'Celcius': 30.21,
+        'Fahrenheit': 87
+    }
+    TAGS = {
+        'host': 'server01',
+        'region': 'us-east'
+    }
+    DATABASE.log_data(FIELDS, 'temperature', TAGS)
+    print(DATABASE.read_data(tags=['Celsius', 'Fahrenheit'], measurements=['temperature']))
```

Comment:
I think for this I may just add another parameter for fields, so then it will be:
```python
DATABASE.read_data(tags=None, fields=['Celsius', 'Fahrenheit'], measurements=['temperature'])
```
this should provide a better distinction

Reply:
Thats a good solution